dbt/tests/generic/test_row_values_match_after_join.sql

```diff
@@ -87,18 +87,26 @@
     {%- endif -%}
 
     select
-        {{ group_by_csv }},
+        {%- if group_by is defined %}
+            {{ group_by_csv }},
+            array_agg({{ model_col }}) as {{ column_alias }},
+            array_agg({{ external_model_col }}) as {{ external_column_alias }}
+        {%- else %}
+            {{ model_col }} as {{ column_alias }},
+            {{ external_model_col }} as {{ external_column_alias }}
+        {%- endif %}
         {% if additional_select_columns_csv -%}
-            {{ additional_select_columns_csv }},
+            , {{ additional_select_columns_csv }}
         {% endif %}
-        array_agg({{ model_col }}) as {{ column_alias }},
-        array_agg({{ external_model_col }}) as {{ external_column_alias }}
     from
         {{ external_model }} as external_model
     {{ join_type }} join -- fmt: off
         (select * from {{ model }}) as model {{ join_condition }}
+
+    {%- if group_by is defined %}
```

Comment:
**[Question, non-blocking]** If `group_by` is not defined, but the test doesn't implement a `join_condition` that specifically filters for mismatching rows (as the one in this PR does), then am I right that the test won't catch mismatching rows due to the `having` clause being excluded? That works, but it requires the caller to understand the implementation details of the generic to know that it won't automatically filter for mismatching rows, so I wonder if there's a way we could make the interface friendlier in that case.

Reply:
Good catch! I think defaulting to a WHERE clause here solves this, see c4f473f.